07-documentation.md

```diff
@@ -0,0 +1,80 @@
+# REST documentation
+
+With the separation of frontend and backend it has become of critical importance that the REST interface is clearly documented. It is possible that frontend and backend tasks are performed by different developers, which simply means that the frontender should know what requests to call and what output to expect.
+
+Besides an overview of resources the documentation should also discuss cross cutting concerns, such as status codes, error handling and security. The REST document serves as contract, helping consumers of the interface and ensuring that the interface stays consistent and clear.
```

Comment:
Cross-cutting concerns are technical aspects, which are required for several functional methods. Typically aspect-oriented programming is used to insert this logic. A couple of examples of cross-cutting concerns are writing log statements about the duration of a method call, or the conversion of a NotFoundException to a 404 error. These cross-cutting concerns are best handled in a generic way.